### PR TITLE
Direct deposit definition

### LINKED ISSUE #15843: Development for direct deposit component
## Issue Description
Development for direct deposit component 

---
## Tasks
- [ ] Take artifact from design and implement 
- [ ] Unit tests
- [ ] Stories/description of what the component does 

## Acceptance Criteria
- [ ] Sign off from designer
---
## How to configure this issue
- [ ] **Attached to a Milestone** (when will this be completed?)
- [ ] **Attached to an Epic** (what body of work is this a part of?)
- [ ] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `service-design`, `tools-be`, `tools-fe`)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [ ] **Labeled with Type** (`bug`, `request`, `discovery`, `documentation`, etc.)

### PR DESCRIPTION
**Pull request by cvalarida**
## Description
Closes https://github.com/department-of-veterans-affairs/va.gov-team/issues/15843

## Testing done
- Wrote unit tests
- Created a new form and used it there to verify visually

## Screenshots

<details>
<summary>All the fields</summary>

image
</details>

<details>
<summary>Without the optional fields</summary>

image
</details>

<details>
<summary>Direct deposit declined</summary>

image
</details>

<details>
<summary>Additional info content</summary>

image
</details>

<details>
<summary>Pre-filled bank information</summary>

image
</details>

<details>
<summary>Edited bank information</summary>

image
</details>

## Acceptance criteria
- [x] Bank name is optional
- [x] Decline direct deposit is optional
- [x] All the fields appear in the correct order
- [x] Declining the direct deposit hides the rest of the fields

### CODE REVIEW

File: src/platform/forms-system/src/js/definitions/content/directDeposit.jsx
Comment (CrystabelReiter) [inline src/platform/forms-system/src/js/definitions/content/directDeposit.jsx:10]: Can we remove the sentence "If you're approved for the Rogers STEM Scholarship, we'll need to know where to deposit the funds." @cvalarida

**Review by bkjohnson — COMMENTED** [review]
I need to understand fields/widgets a bit more :upside_down_face:

File: src/platform/forms-system/src/js/definitions/content/directDeposit.jsx
Comment (cvalarida) [reply src/platform/forms-system/src/js/definitions/content/directDeposit.jsx:10]: :joy: Yeah, that...yeah. :see_no_evil: :sweat_smile:

File: src/platform/forms-system/src/js/definitions/content/directDeposit.jsx
Comment (cvalarida) [reply src/platform/forms-system/src/js/definitions/content/directDeposit.jsx:41]: I don't think so. The "extra" info attached to the GA event is going to be different, but I think that's fine.

Now that you mention it, though, I need to remove that `edu-10203--` :sweat_smile:

File: src/platform/forms-system/src/js/definitions/directDeposit.js
Comment (cvalarida) [reply src/platform/forms-system/src/js/definitions/directDeposit.js:158]: Good call. I think this is an artifact from a different approach. I was going to make the argument passed into the function more than just optional fields, but turns out I didn't need to.

Removing!

File: src/platform/forms-system/src/js/definitions/directDeposit.js
Comment (cvalarida) [reply src/platform/forms-system/src/js/definitions/directDeposit.js:112]: Maybe? I honestly don't know why it is this way. I just copied it from another schema. :see_no_evil: 

I'm not against making it a number, but I don't know what the ramifications of that would be, so I'm not especially comfortable with it in this PR.

File: src/platform/forms-system/src/js/definitions/directDeposit.js
Comment (cvalarida) [reply src/platform/forms-system/src/js/definitions/directDeposit.js]: 100% could! Good call.

File: src/platform/forms-system/src/js/definitions/content/directDeposit.jsx
Comment (cvalarida) [reply src/platform/forms-system/src/js/definitions/content/directDeposit.jsx:30]: Hmm...I just copied this from the 10203. @1Copenut thoughts?

File: src/platform/forms-system/src/js/definitions/directDeposit.js
Comment (cvalarida) [reply src/platform/forms-system/src/js/definitions/directDeposit.js:51]: It's not an actual hook. Just a function that starts with `use`. :man_shrugging:

**Review by bkjohnson — APPROVED** [review]
Seems fine to me :+1:

**Comment by cvalarida** [comment]
**Status update:** Waiting on content review from @DanielleThierryUSDSVA

File: src/platform/forms-system/src/js/definitions/content/directDeposit.jsx
Comment (bkjohnson) [inline src/platform/forms-system/src/js/definitions/content/directDeposit.jsx:39]: Are the brackets inside the string part of the mockup? Is this something that we want/expect veterans to see?

**Comment by cvalarida** [comment]
Updated content:
image

The link opens https://veteransbenefitsbanking.org/ in a new tab.

**Review by CrystabelReiter — APPROVED** [review]

File: src/platform/forms-system/test/js/components/PaymentView.unit.spec.jsx
Comment (bkjohnson) [inline src/platform/forms-system/test/js/components/PaymentView.unit.spec.jsx:30]: Yer a wizard!

**Review by bkjohnson — APPROVED** [review]

**Pull request merged**
